### ISSUE TITLE
Infinite loop in walkTNodeTree with nested *ngIf

### ISSUE THREAD

**Issue by mrnateriver** [angular/angular#29700]
I've only just recently started using Angular (v7), so this might be my fault, but my experience tells me otherwise.

# 🐞 bug report

### Affected Package
@angular/core 7.2.10

### Description
```
<ng-container id="c1" *ngIf="b === 1">   <el *ngIf="a === 1"> <el *ngIf="a !== 1"> </ng-container>
<ng-container id="c2" *ngIf="b !== 1">   <el *ngIf="a === 1"> <el *ngIf="a !== 1"> </ng-container>
```
switching `b` from 1 to 2 while 'a === 1' causes infinite loop in `walkTNodeTree` (`core/fesm2015/core.js:4777`) and from the looks of it, it happens because upon walking "past" the first element in #c1, `currentView[NEXT]` is always null (since neither `currentView` nor `NEXT` change) and `tnode.next` is null because of the other condition (`a`).

However, if I move `<el>` outside of `<ng-container>` and replace their conditions with `b === 1 && ...` then everything works.

## 🔬 Minimal Reproduction
Couldn't reproduce it on StackBlitz, so here is the whole repo of the project I'm working on:
https://github.com/mrnateriver/angular-temporary-repo
(sorry that it's in russian, but you can just do `ng serve`, navigate to `/login` and click the yellow button on the right ("Далее")).

The component in question is `src/app/features/auth/login-form`, and it is embedded in `login-page`.

## 🌍  Your Environment
<pre><code>
Angular CLI: 7.3.6
Node: 10.15.3
OS: win32 x64
Angular: 7.2.10
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.13.6
@angular-devkit/build-angular     0.13.6
@angular-devkit/build-optimizer   0.13.6
@angular-devkit/build-webpack     0.13.6
@angular-devkit/core              7.3.6
@angular-devkit/schematics        7.3.6
@angular/cdk                      7.3.5
@angular/cli                      7.3.6
@angular/material                 7.3.5
@ngtools/webpack                  7.3.6
@schematics/angular               7.3.6
@schematics/update                0.13.6
rxjs                              6.3.3
typescript                        3.2.4
webpack                           4.29.0
</code></pre>

Tested in Chrome 73.0.3683.86 and Firefox 66.0.2.

## P.S.
Even if it's not a bug but rather I've done something wrong, I still think that loop shouldn't be infinite in any case.

**Comment by ocombe** [comment]
I tried your example and confirmed the issue.
Then I tried it with the latest beta and everything seems to work fine.
Here is the list of dependencies if you want to try:
```
"dependencies": {
    "@angular/animations": "8.0.0-beta.13",
    "@angular/cdk": "8.0.0-beta.2",
    "@angular/common": "8.0.0-beta.13",
    "@angular/compiler": "8.0.0-beta.13",
    "@angular/core": "8.0.0-beta.13",
    "@angular/forms": "8.0.0-beta.13",
    "@angular/material": "8.0.0-beta.2",
    "@angular/platform-browser": "8.0.0-beta.13",
    "@angular/platform-browser-dynamic": "8.0.0-beta.13",
    "@angular/router": "8.0.0-beta.13",
    "core-js": "^2.5.4",
    "hammerjs": "^2.0.8",
    "http-status-codes": "^1.3.2",
    "rxjs": "~6.3.3",
    "tslib": "^1.9.0",
    "zone.js": "~0.8.26"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "0.800.0-beta.15",
    "@angular/cli": "8.0.0-beta.15",
    "@angular/compiler-cli": "8.0.0-beta.13",
    "@angular/language-service": "8.0.0-beta.13",
    "@types/jasmine": "^3.3.9",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "~4.5.0",
    "jasmine-core": "~3.3.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.11.0",
    "typescript": "3.4.3"
  }
```

I'll close it since it works now, let me know if you still have an issue.

**Comment by pkozlowski-opensource** [comment]
> I'll close it since it works now, let me know if you still have an issue.

@ocombe maybe it would be a good idea to add an integration TestBed test for this exact case since it seems like our current test coverage is not enough to catch it?

**Comment by ocombe** [comment]
It'll be hard to write a test for a bug that doesn't exist anymore, I'm not really sure what it was to begin with

**Comment by pkozlowski-opensource** [comment]
> It'll be hard to write a test for a bug that doesn't exist anymore, I'm not really sure what it was to begin with

I was thinking of adding a test that has a component with a template from this issue and asserts things are rendered as expected.

**Comment by speige** [comment]
+1 I'm getting this using the latest versions of everything. I'll see if I can make a StackBlitz.

**Comment by speige** [comment]
I was able to fix my issue.

Old Code:
```
	<mat-tab-group>
		<mat-tab>
			<div ngIf="">
			</div>
		</mat-tab>
	</mat-tab-group>
```

New Code:
```
	<mat-tab-group>
		<mat-tab ngIf="">
			<div>
			</div>
		</mat-tab>
	</mat-tab-group>
```

It'd be nice if the walkTNodeTree didn't have an infinite loop and if it gave a console error if something's wrong.

Strangely, I tried to copy a minimal reproduction into StackBlitz & couldn't get reproduce it. Maybe ivy-ngcc wasn't running or maybe there's something more else in my project that's causing the issue.
